src/components/SliderField/SliderField.tsx

```diff
@@ -0,0 +1,204 @@
+    to: SliderDefault.TO,
+    lastInteraction: SliderLastInteraction.from
+  }
+
+  componentDidMount(): void {
+    if (this.props.defaultValue === undefined) {
+      this.setState((prevState) => {
+        return {
+          ...prevState,
+          from: !isNaN(this.props.min) ? this.props.min : prevState.from,
+          to: !isNaN(this.props.max) ? this.props.max : prevState.to
+        }
+      })
+    } else {
+      if (Array.isArray(this.props.defaultValue)) {
+        this.setState({
+          from: this.props.defaultValue[0],
+          to: this.props.defaultValue[1]
+        })
+      } else {
+        this.setState({
+          to: Number(this.props.defaultValue)
+        })
+      }
+    }
+  }
+
+  render(): JSX.Element {
+    const { header, className, onChange, onMouseUp, range } = this.props
+
+    const min = this.props.min || SliderDefault.MIN
+    const max = this.props.max || SliderDefault.MAX
+
+    const classes = ['dcl', 'sliderfield']
+    if (className) {
+      classes.push(className)
+    }
+
+    const handleChangeFrom = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const input = e.target
+
+      this.setState(
+        (prevState) => {
+          return {
+            ...prevState,
+            from: Math.min(Number(input.value), prevState.to),
+            lastInteraction: SliderLastInteraction.from
+          }
+        },
+        () => {
+          if (onChange) {
+            onChange(e, [this.state.from, this.state.to])
+          }
+        }
+      )
+    }
+
+    const handleChangeTo = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const input = e.target
+
+      this.setState(
+        (prevState) => {
+          return {
+            ...prevState,
+            to: Math.max(Number(input.value), prevState.from),
+            lastInteraction: SliderLastInteraction.to
+          }
+        },
+        () => {
+          if (onChange) {
+            onChange(
+              e,
+              range ? [this.state.from, this.state.to] : this.state.to
+            )
+          }
+        }
+      )
+    }
+
+    const handleMouseUp = (e: React.MouseEvent<HTMLInputElement>) => {
+      if (onMouseUp) {
+        onMouseUp(e, range ? [this.state.from, this.state.to] : this.state.to)
+      }
+    }
+
+    const diffMaxMin = max - min
+    const trackLeft = range
+      ? `${((this.state.from - min) * 100) / diffMaxMin}%`
+      : 0
+    const trackRight = range
+      ? `${((max - this.state.to) * 100) / diffMaxMin}%`
+      : `${100 - ((this.state.to - min) * 100) / diffMaxMin}%`
+    const handleFrom = `${((this.state.from - min) * 100) / diffMaxMin}%`
+    const handleTo = `${((this.state.to - min) * 100) / diffMaxMin}%`
+
+    const label =
+      this.props.label ||
+      (range ? `${this.state.from} - ${this.state.to}` : this.state.to)
+
+    return (
+      <div>
+        <div className={classes.join(' ')}>
+          <div className={'dcl sliderfield-header'}>{header}</div>
```

Comment:
we can change these clases to be a bit more explicit and consistent!

```suggestion
          <div className="dcl sliderfield-header">{header}</div>
```

same for the other `className`s below

Reply:
good observation